src/main/java/com/flagsmith/FlagsmithClient.java

```diff
@@ -192,17 +197,23 @@ private Flags getIdentityFlagsFromDocument(String identifier, Map<String, Object
 
     return Flags.fromFeatureStateModels(
         featureStates,
-        flagsmithSdk.getConfig().getAnalyticsProcessor(),
+        getConfig().getAnalyticsProcessor(),
         identity.getCompositeKey(),
-        flagsmithSdk.getConfig().getFlagsmithFlagDefaults());
+        getConfig().getFlagsmithFlagDefaults());
   }
 
-  private Flags getEnvironmentFlagsFromApi() throws FlagsmithApiError {
+  private Flags getEnvironmentFlagsFromApi() throws FlagsmithApiError, FlagsmithClientError {
     try {
       return flagsmithSdk.getFeatureFlags(Boolean.TRUE);
     } catch (Exception e) {
-      if (flagsmithSdk.getConfig().getFlagsmithFlagDefaults() != null) {
+      if (getConfig().getFlagsmithFlagDefaults() != null) {
         return getDefaultFlags();
+      } else if (environment != null) {
+        try {
+          return getEnvironmentFlagsFromDocument();
+        } catch (FlagsmithClientError ce) {
+          // Do nothing and fall through to FlagsmithApiError
```

Comment:
Might be my limited knowledge of Java, but if we're catching `FlagsmithClientError` here, why did we need to update the function definition?

Reply:
You're correct, I've removed it from the the `throws`.